### PR TITLE
Add a comment and do light cleanup in EncounterForm

### PR DESCRIPTION
**Pull request by brandones**

### CODE REVIEW

File: src/components/form/EncounterForm.jsx [hunk 1]
```diff
@@ -155,27 +151,21 @@ const mapStateToProps = (state, props) => {
         formInstanceId: props.formInstanceId,
         patient: props.patient,
         encounter: props.encounter,
-        encounterRole: props.encounterRole ? props.encounterRole : null,
+        encounterRole: props.encounterRole || props.encounterRole || null,
```
Comment (brandones) [inline src/components/form/EncounterForm.jsx:154]: The `|| null`s probably shouldn't be necessary, but who knows, maybe something somewhere downstream differentiates between `""` and `null`

Comment (mogoodrich) [reply src/components/form/EncounterForm.jsx:154]: why the double props.encounterRole?

File: src/components/form/EncounterForm.jsx [hunk 2]
```diff
@@ -155,27 +151,21 @@ const mapStateToProps = (state, props) => {
...
         encounterType: props.encounterType,
-        location: props.location ? props.location :
-          sessionLocation ? sessionLocation : null,
+        location: props.location || sessionLocation || null,
```
Comment (mogoodrich) [inline src/components/form/EncounterForm.jsx:156]: this uses the session location variable that you removed above?

File: src/components/form/EncounterForm.jsx [hunk 3]
```diff
@@ -155,27 +151,21 @@ const mapStateToProps = (state, props) => {
...
         manuallyExitSubmitMode: props.manuallyExitSubmitMode,
         orderForObs: props.orderForObs,
-        provider: props.provider ? props.provider :
-          currentProvider ? currentProvider : null,
+        provider: props.provider || currentProvider || null,
```
Comment (mogoodrich) [inline src/components/form/EncounterForm.jsx:159]: this uses the current provider variable that you removed above?

File: src/components/form/EncounterForm.jsx (hunk 2 above)
Comment (brandones) [reply src/components/form/EncounterForm.jsx:160]: I put it back

File: src/components/form/EncounterForm.jsx (hunk 3 above)
Comment (brandones) [reply src/components/form/EncounterForm.jsx:163]: I put it back

**Comment by brandones** [comment]
@mogoodrich Whoops, sorry for the really sloppy work on a nearly trivial PR. I should have self-reviewed before. I fixed it up.

**Pull request merged**